Source/Evolve/WarpXEvolve.cpp

```diff
@@ -209,6 +213,17 @@ WarpX::Evolve (int numsteps)
             mypc->SortParticlesByBin(sort_bin_size);
         }
 
+        if( do_electrostatic != ElectrostaticSolverAlgo::None ) {
+            // Electrostatic solver:
+            // For each species: deposit charge and add the associated space-charge
+            // E and B field to the grid ; this is done at the end of the PIC
+            // loop (i.e. immediately after a `Redistribute` and before particle
+            // positions are next pushed) so that the particles do not deposit out of bounds
+            // and so that the fields are at the correct time in the output.
+            bool const reset_fields = true;
+            ComputeSpaceChargeField( reset_fields );
+        }
```

Comment:
This makes sense.
As a result, I think that almost every operation done in the function `OneStep_nosub` is now superfluous for electrostatic (except for the particle push of the position and momentum). 
Should we therefore avoid calling `OneStep_nosub` and instead have a `OneStep_nosub_electrostatic` that incorporates the above code but adds the particle push?
(No need to do it directly in this PR ; I just wanted to check that I undertstood correctly.)

Reply:
Yes, that is basically correct. The `OneStep_nosub` also does field ionization and QED, but those are probably not of interest in electrostatic. It also does resampling which might be useful. Making that change would be part of a bigger clean up, for example which turns off the current deposition.

Reply:
Sounds good, thanks!